common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -51,108 +58,113 @@ public void testCreateBasicListener() throws InterruptedException {
-    @Ignore("Test is flaky with 220ms delay")
-    @Test
-    public void testAddChangeToBasicLiveQuery() throws CouchbaseLiteException, InterruptedException {
-        final Query query = QueryBuilder
-            .select(SelectResult.expression(Meta.id))
-            .from(DataSource.database(baseTestDb))
-            .where(Expression.property(KEY).greaterThanOrEqualTo(Expression.intValue(0)))
-            .orderBy(Ordering.property(KEY).ascending());
+        ListenerToken token2 = query.addChangeListener(
+            testSerialExecutor,
+            change -> {
+                latch2[atmCount.get(1)].countDown();
+                atmCount.set(1, atmCount.get(1) + 1);
+            });
 
-        final CountDownLatch latch = new CountDownLatch(1);
+        try {
+            // listener should get notify after query run time
+            assertTrue(latch2[0].await(TOLERABLE_QUERY_RUN_TIME_MS, TimeUnit.MILLISECONDS));
 
-        ListenerToken token = query.addChangeListener(testSerialExecutor, change -> latch.countDown());
+            // creation of the second listener should not trigger first listener callback
+            assertFalse(latch1[1].await(TOLERABLE_QUERY_RUN_TIME_MS * 2, TimeUnit.MILLISECONDS));
 
-        createDocNumbered(10);
+            createDocNumbered(11);
 
-        try { assertTrue(latch.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS)); }
-        finally { query.removeChangeListener(token); }
+            // introducing change in database should trigger both listener callbacks after the time it takes
+            // for core to wait and rerun query
+            assertTrue(latch1[1].await(TOLERABLE_DATABASE_CHANGE_DELAY_MS, TimeUnit.MILLISECONDS));
+            assertTrue(latch2[1].await(TOLERABLE_DATABASE_CHANGE_DELAY_MS, TimeUnit.MILLISECONDS));
+        }
+        finally {
+            query.removeChangeListener(token1);
+            query.removeChangeListener(token2);
+        }
     }
 
-    // !!! What the ever-luvvin heck is this trying to test???
-    @Ignore("Test fail with 220ms delay")
+    // When a result set is closed, we should still be able to introduce a change
+    @SuppressWarnings("unused")
     @Test
     public void testCloseResultsInLiveQueryListener() throws CouchbaseLiteException, InterruptedException {
         final Query query = QueryBuilder
             .select(SelectResult.expression(Meta.id))
             .from(DataSource.database(baseTestDb));
 
-        final int[] count = new int[] {0};
+        final AtomicIntegerArray atmCount = new AtomicIntegerArray(1);
         final CountDownLatch[] latches = new CountDownLatch[2];
         for (int i = 0; i < latches.length; i++) { latches[i] = new CountDownLatch(1); }
+
         ListenerToken token = query.addChangeListener(
             testSerialExecutor,
             change -> {
-                // this will close the resultset
+                //this try-with resources will close the rs
```

Comment:
Yeah... right... but the whole point of this test is closing the resultSet.  What would you think of:
change -> {
    change.getResults().close()
    latches[atmCount.getAndIncrement(0)].countDown();
}